src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/ThunkPool.cs

```diff
@@ -44,8 +44,8 @@ internal static class Constants
         public static readonly int ThunkCodeSize = RuntimeImports.RhpGetThunkSize();
         public static readonly int NumThunksPerBlock = RuntimeImports.RhpGetNumThunksPerBlock();
         public static readonly int NumThunkBlocksPerMapping = RuntimeImports.RhpGetNumThunkBlocksPerMapping();
-        public static readonly uint ThunkBlockSize = (uint)RuntimeImports.RhpGetThunkBlockSize();
-        public static readonly nuint ThunkBlockSizeMask = ThunkBlockSize - 1;
+        public static readonly uint ThunkDataBlockSize = (uint)RuntimeImports.RhpGetThunkDataBlockSize();
+        public static readonly nuint ThunkDataBlockSizeMask = ThunkDataBlockSize - 1;
```

Comment:
```suggestion
        public static readonly uint ThunkDataBlockSize = (uint)(ThunkDataSize * NumThunksPerBlock + IntPtr.Size);
        public static readonly nuint ThunkDataBlockSizeMask = BitOperations.RoundUpToPowerOf2(ThunkDataBlockSize) - 1;
```
Can this be inferred from the existing values instead of introducing method that needs to be implemented for every architecture?

Reply:
It is better to have less arch-specific code. ThunkPool has more arch-specific code than it needs. We should not be adding more.

My suggestion implements the layout described at the top of this file. I do not think that it leaks any arch-specific details.

Reply:
ok, for this case we could calculate it in managed build on the current data points we already have from arch implementation, but would be nice to clean this whole thing up at some point in time.